### PR TITLE
Add SearchBar to the event viewer

### LINKED ISSUE #3930: Add search bar in the event viewer and stdout/stderr views
**Is your feature request related to a problem? Please describe.**
As a user I would like to be able to quickly find information in different logs (event viewer, stdout/stderr from jobs, etc).

**Describe the solution you'd like**
I would like a search bar that allows me to search in text content

**Describe alternatives you've considered**
N/A

**Additional context**
If there are other view with large amounts of text (in addition to event viewer and stdout/stderr jobs) these should also have the same search feature.

### PR DESCRIPTION
**Pull request by DanSava**
The SearchBar will help search and highlight terms in large text bodies.

**Issue**
Resolves #3930

(Screenshot of new behavior in GUI if applicable)

image

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: tests/unit_tests/gui/ertwidgets/test_search_box.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+from qtpy.QtGui import QColor, QTextCursor
+from qtpy.QtWidgets import QApplication, QPlainTextEdit
+
+from ert.gui.tools.search_bar import SearchBar
+
+
+def test_search_box_highlights_matching_text():
```
Comment (xjules) [inline tests/unit_tests/gui/ertwidgets/test_search_box.py:7]: Just wondering if all these tests could be tackled with one parametrize?

Comment (DanSava) [reply tests/unit_tests/gui/ertwidgets/test_search_box.py:7]: Sure I can combine them in one test

Comment (DanSava) [reply tests/unit_tests/gui/ertwidgets/test_search_box.py]: Done

**Review by eivindjahren — CHANGES_REQUESTED** [review]
I think it would make sense to have the search bar on top

**Review by eivindjahren — APPROVED** [review]

**Pull request merged**